public_html/admin/controller/responses/listing_grid/customer.php

```diff
@@ -170,13 +170,14 @@ public function update()
         $this->loadLanguage('sale/customer');
         if (!$this->user->canModify('listing_grid/customer')) {
             $error = new AError('');
-            return $error->toJSONResponse(
+            $error->toJSONResponse(
                 'NO_PERMISSIONS_402',
                 [
                     'error_text'  => sprintf($this->language->get('error_permission_modify'), 'listing_grid/customer'),
                     'reset_value' => true,
                 ]
             );
+            return;
```

Comment:
Why error is no longer returned? Validation should return the error. At least validation has to return FALSE if failed.

Reply:
it's no a validation. Function name is $error->toJSONResponse

Reply:
it's a void function. It send response as json data directly to php output. No return data needed here.